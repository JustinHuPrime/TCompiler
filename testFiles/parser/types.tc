module foo;

int a;
int const b;
int volatile c;
int['a'] d;
int *e;
int(int, int arg2) f;
ubyte const volatile ub1;
ubyte volatile const ub2;
ubyte const[1] const arry;

void bar() {
  int a;
  int const b;
  int volatile c;
  int['a'] d;
  int *e;
  int(int, int arg2) f;
  ubyte const volatile ub1;
  ubyte volatile const ub2;
  ubyte const[1] const arry;
}