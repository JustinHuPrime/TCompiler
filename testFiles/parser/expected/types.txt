testFiles/parser/input/types.tc (code):
FILE(1, 1, STAB(ENTRY(e, VARIABLE(testFiles/parser/input/types.tc, 7, 6, int *)), ENTRY(d, VARIABLE(testFiles/parser/input/types.tc, 6, 10, int[97])), ENTRY(f, VARIABLE(testFiles/parser/input/types.tc, 8, 20, int(int, int))), ENTRY(a, VARIABLE(testFiles/parser/input/types.tc, 3, 5, int)), ENTRY(arry, VARIABLE(testFiles/parser/input/types.tc, 11, 22, ubyte const[1] const)), ENTRY(c, VARIABLE(testFiles/parser/input/types.tc, 5, 14, int volatile)), ENTRY(b, VARIABLE(testFiles/parser/input/types.tc, 4, 11, int const)), ENTRY(bar, FUNCTION(testFiles/parser/input/types.tc, 13, 1, void())), ENTRY(ub1, VARIABLE(testFiles/parser/input/types.tc, 9, 22, ubyte volatile const)), ENTRY(ub2, VARIABLE(testFiles/parser/input/types.tc, 10, 22, ubyte volatile const))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), VARDEFN(3, 1, KEYWORDTYPE(3, 1, int), ID(3, 5, a, REFERENCES(testFiles/parser/input/types.tc, 3, 5)), (null)), VARDEFN(4, 1, MODIFIEDTYPE(4, 1, CONST, KEYWORDTYPE(4, 1, int)), ID(4, 11, b, REFERENCES(testFiles/parser/input/types.tc, 4, 11)), (null)), VARDEFN(5, 1, MODIFIEDTYPE(5, 1, VOLATILE, KEYWORDTYPE(5, 1, int)), ID(5, 14, c, REFERENCES(testFiles/parser/input/types.tc, 5, 14)), (null)), VARDEFN(6, 1, ARRAYTYPE(6, 1, KEYWORDTYPE(6, 1, int), LITERAL(6, 5, CHAR('a'))), ID(6, 10, d, REFERENCES(testFiles/parser/input/types.tc, 6, 10)), (null)), VARDEFN(7, 1, MODIFIEDTYPE(7, 1, POINTER, KEYWORDTYPE(7, 1, int)), ID(7, 6, e, REFERENCES(testFiles/parser/input/types.tc, 7, 6)), (null)), VARDEFN(8, 1, FUNPTRTYPE(8, 1, KEYWORDTYPE(8, 1, int), KEYWORDTYPE(8, 1, int), KEYWORDTYPE(8, 1, int)), ID(8, 20, f, REFERENCES(testFiles/parser/input/types.tc, 8, 20)), (null)), VARDEFN(9, 1, MODIFIEDTYPE(9, 1, VOLATILE, MODIFIEDTYPE(9, 1, CONST, KEYWORDTYPE(9, 1, ubyte))), ID(9, 22, ub1, REFERENCES(testFiles/parser/input/types.tc, 9, 22)), (null)), VARDEFN(10, 1, MODIFIEDTYPE(10, 1, CONST, MODIFIEDTYPE(10, 1, VOLATILE, KEYWORDTYPE(10, 1, ubyte))), ID(10, 22, ub2, REFERENCES(testFiles/parser/input/types.tc, 10, 22)), (null)), VARDEFN(11, 1, MODIFIEDTYPE(11, 1, CONST, ARRAYTYPE(11, 1, MODIFIEDTYPE(11, 1, CONST, KEYWORDTYPE(11, 1, ubyte)), LITERAL(11, 13, UBYTE(1)))), ID(11, 22, arry, REFERENCES(testFiles/parser/input/types.tc, 11, 22)), (null)), FUNDEFN(13, 1, KEYWORDTYPE(13, 1, void), ID(13, 6, bar, REFERENCES(testFiles/parser/input/types.tc, 13, 1)), STAB(), COMPOUNDSTMT(13, 12, STAB(ENTRY(e, VARIABLE(testFiles/parser/input/types.tc, 18, 8, int *)), ENTRY(d, VARIABLE(testFiles/parser/input/types.tc, 17, 12, int[97])), ENTRY(f, VARIABLE(testFiles/parser/input/types.tc, 19, 22, int(int, int))), ENTRY(a, VARIABLE(testFiles/parser/input/types.tc, 14, 7, int)), ENTRY(arry, VARIABLE(testFiles/parser/input/types.tc, 22, 24, ubyte const[1] const)), ENTRY(c, VARIABLE(testFiles/parser/input/types.tc, 16, 16, int volatile)), ENTRY(b, VARIABLE(testFiles/parser/input/types.tc, 15, 13, int const)), ENTRY(ub1, VARIABLE(testFiles/parser/input/types.tc, 20, 24, ubyte volatile const)), ENTRY(ub2, VARIABLE(testFiles/parser/input/types.tc, 21, 24, ubyte volatile const))), VARDEFNSTMT(14, 3, KEYWORDTYPE(14, 3, int), ID(14, 7, a, REFERENCES(testFiles/parser/input/types.tc, 14, 7)), (null)), VARDEFNSTMT(15, 3, MODIFIEDTYPE(15, 3, CONST, KEYWORDTYPE(15, 3, int)), ID(15, 13, b, REFERENCES(testFiles/parser/input/types.tc, 15, 13)), (null)), VARDEFNSTMT(16, 3, MODIFIEDTYPE(16, 3, VOLATILE, KEYWORDTYPE(16, 3, int)), ID(16, 16, c, REFERENCES(testFiles/parser/input/types.tc, 16, 16)), (null)), VARDEFNSTMT(17, 3, ARRAYTYPE(17, 3, KEYWORDTYPE(17, 3, int), LITERAL(17, 7, CHAR('a'))), ID(17, 12, d, REFERENCES(testFiles/parser/input/types.tc, 17, 12)), (null)), VARDEFNSTMT(18, 3, MODIFIEDTYPE(18, 3, POINTER, KEYWORDTYPE(18, 3, int)), ID(18, 8, e, REFERENCES(testFiles/parser/input/types.tc, 18, 8)), (null)), VARDEFNSTMT(19, 3, FUNPTRTYPE(19, 3, KEYWORDTYPE(19, 3, int), KEYWORDTYPE(19, 7, int), KEYWORDTYPE(19, 12, int)), ID(19, 22, f, REFERENCES(testFiles/parser/input/types.tc, 19, 22)), (null)), VARDEFNSTMT(20, 3, MODIFIEDTYPE(20, 3, VOLATILE, MODIFIEDTYPE(20, 3, CONST, KEYWORDTYPE(20, 3, ubyte))), ID(20, 24, ub1, REFERENCES(testFiles/parser/input/types.tc, 20, 24)), (null)), VARDEFNSTMT(21, 3, MODIFIEDTYPE(21, 3, CONST, MODIFIEDTYPE(21, 3, VOLATILE, KEYWORDTYPE(21, 3, ubyte))), ID(21, 24, ub2, REFERENCES(testFiles/parser/input/types.tc, 21, 24)), (null)), VARDEFNSTMT(22, 3, MODIFIEDTYPE(22, 3, CONST, ARRAYTYPE(22, 3, MODIFIEDTYPE(22, 3, CONST, KEYWORDTYPE(22, 3, ubyte)), LITERAL(22, 15, UBYTE(1)))), ID(22, 24, arry, REFERENCES(testFiles/parser/input/types.tc, 22, 24)), (null)))))
